crates/viewer/re_time_panel/src/data_density_graph.rs

```diff
@@ -366,6 +367,206 @@ fn smooth(density: &[f32]) -> Vec<f32> {
+        ) else {
+            return;
+        };
+
+        density_graph.add_range((min_x, max_x), count as _);
+
+        if let Some(pointer_pos) = pointer_pos {
+            let is_hovered = if (max_x - min_x).abs() < 1.0 {
+                // Are we close enough to center?
+                let center_x = (max_x + min_x) / 2.0;
+                let distance_sq = pos2(center_x, center_y).distance_sq(pointer_pos);
+
+                /*
+                let is_hovered = distance_sq < interact_radius_sq;
+                let color = if is_hovered {
+                    egui::Color32::LIGHT_YELLOW
+                } else {
+                    egui::Color32::YELLOW
+                };
+                ui.ctx().debug_painter().debug_rect(
+                    egui::Rect::from_center_size(
+                        egui::pos2(center_x, center_y),
+                        egui::vec2(4.0, 4.0),
+                    ),
+                    color,
+                    "",
+                );
+
+                is_hovered
+                */
+
+                distance_sq < interact_radius_sq
+            } else {
+                // Are we within time range rect?
+                let time_range_rect = Rect {
+                    min: egui::pos2(min_x, row_rect.min.y),
+                    max: egui::pos2(max_x, row_rect.max.y),
+                };
+
+                /*
+                let is_hovered = time_range_rect.contains(pointer_pos);
+                let color = if is_hovered {
+                    egui::Color32::LIGHT_YELLOW
+                } else {
+                    egui::Color32::YELLOW
+                };
+                ui.ctx()
+                    .debug_painter()
+                    .debug_rect(time_range_rect, color, "");
+
+                is_hovered
+                */
+
+                time_range_rect.contains(pointer_pos)
+            };
+
+            if is_hovered {
+                hovered_time_range = hovered_time_range.union(time_range);
+                num_hovered_messages += count;
+            }
+        }
+    };
+
+    /* let mut num_chunks = 0; */
+    if let Some(component_name) = item.component_name {
+        let chunks = db
+            .store()
+            .range_relevant_chunks(&query, &item.entity_path, component_name);
+
+        /* num_chunks += chunks.len(); */
+
+        for chunk in chunks {
+            let Some(events) = chunk.num_events_for_component(component_name) else {
+                continue;
+            };
+
+            let Some(chunk_timeline) = chunk.timelines().get(&timeline) else {
+                continue;
+            };
+
+            add_data_point(chunk_timeline.time_range(), events);
+        }
+    } else if let Some(components) = store.all_components(&timeline, &item.entity_path) {
+        for component_name in components {
+            let chunks =
+                db.store()
+                    .range_relevant_chunks(&query, &item.entity_path, component_name);
```

Comment:
This will likely return the same chunks many times, e.g. if you log a point cloud with both color and positions, it will go into the same chunk, so we need to deduplicate the chunks. Probably nicest to do so in some helper functions in `ChunkStore` so we just call `range_relevant_chunks_for_entity` here.